base/src/com/google/idea/blaze/base/lang/buildfile/references/ExternalWorkspaceReferenceFragment.java

```diff
@@ -42,26 +50,45 @@ public ExternalWorkspaceReferenceFragment(LabelReference labelReference) {
   @Override
   public TextRange getRangeInElement() {
     String rawText = myElement.getText();
-    boolean valid = LabelUtils.getExternalWorkspaceComponent(myElement.getStringContents()) != null;
-    if (!valid) {
+    String unquotedText = LabelUtils.trimToDummyIdentifier(myElement.getStringContents());
+    QuoteType quoteType = myElement.getQuoteType();
+
+    String externalWorkspace = LabelUtils.getExternalWorkspaceComponent(unquotedText);
+    if (!unquotedText.trim().isEmpty() && externalWorkspace == null) {
       return TextRange.EMPTY_RANGE;
     }
+
     int endIndex = rawText.indexOf("//");
     if (endIndex == -1) {
-      endIndex = rawText.length() - 1;
+      endIndex = rawText.length() - quoteType.quoteString.length();
+    } else {
+      endIndex += 2;
     }
     return new TextRange(1, endIndex);
   }
 
   @Nullable
   @Override
-  public FuncallExpression resolve() {
+  public BuildElement resolve() {
     String name = LabelUtils.getExternalWorkspaceComponent(myElement.getStringContents());
     if (name == null) {
       return null;
     }
+
     BuildFile workspaceFile = resolveProjectWorkspaceFile(myElement.getProject());
-    return workspaceFile != null ? workspaceFile.findRule(name) : null;
+    if (workspaceFile != null) {
+      FuncallExpression expression = workspaceFile.findRule(name);
+      if (expression != null) {
+        return expression;
+      }
+    };
+
+    WorkspaceRoot workspaceRoot = WorkspaceHelper.getExternalWorkspace(myElement.getProject(), name);
+    if (workspaceRoot != null) {
+      return BuildReferenceManager.getInstance(myElement.getProject()).findBuildFile(workspaceRoot.directory());
```

Comment:
Yea. This is slightly wrong, too, since it should point to either a `WORKSPACE` or a `MODULE.xx` file anyway. It's a net improvement though and is still helpful for the next step: to root completion of packagePaths from the containing folder